internal/workflow.go

```diff
@@ -383,35 +382,40 @@ func NewFuture(ctx Context) (Future, Settable) {
 	return impl, impl
 }
 
-func (wc *workflowEnvironmentInterceptor) ProcessSignal(Context, string, interface{}) error {
+func (wc *workflowEnvironmentInterceptor) HandleSignal(ctx Context, in *HandleSignalInput) error {
+	eo := getWorkflowEnvOptions(ctx)
+	// We don't want this code to be blocked ever, using sendAsync().
+	ch := eo.getSignalChannel(ctx, in.SignalName).(*channelImpl)
+	if !ch.SendAsync(in.Arg) {
+		return fmt.Errorf("exceeded channel buffer size for signal: %v", in.SignalName)
+	}
 	return nil
 }
 
-func (wc *workflowEnvironmentInterceptor) HandleQuery(_ Context, _ string, args *commonpb.Payloads,
-	handler func(*commonpb.Payloads) (*commonpb.Payloads, error)) (*commonpb.Payloads, error) {
-	return handler(args)
-}
-
-func (wc *workflowEnvironmentInterceptor) ExecuteWorkflow(ctx Context, workflowType string, inputArgs ...interface{}) (results []interface{}) {
-	args := []reflect.Value{reflect.ValueOf(ctx)}
-	for _, arg := range inputArgs {
-		// []byte arguments are not serialized
-		switch arg.(type) {
-		case []byte:
-			args = append(args, reflect.ValueOf(arg))
-		default:
-			args = append(args, reflect.ValueOf(arg).Elem())
+func (wc *workflowEnvironmentInterceptor) HandleQuery(ctx Context, in *HandleQueryInput) (interface{}, error) {
+	eo := getWorkflowEnvOptions(ctx)
+	handler, ok := eo.queryHandlers[in.QueryType]
+	// Should never happen because its presence is checked before this call too
```

Comment:
if the key not present, what error does the check before here returns? we rely on the error message below to tell what are the supported query types in error message.

Reply:
See `RegisterQueryHandler` in `workflow_internal.go` which does this same logic and sends back this same error as it always has before calling the interceptors which trickle down to this call.

Also note that we are adding #601 at some point to stop using error messages for such things.